src/frontend/src/pages/MyGroup/PendingApplications.jsx

```diff
@@ -1,33 +1,93 @@
-                >
-                    <Table {...getTableProps()}>
-                        <SimpleHeader headers={headers} getHeaderProps={getHeaderProps} />
-                        <TableBody>
-                            {rows.map(row => (
-                                <SimpleRow key={row.id} row={row} />
-                            ))}
-                        </TableBody>
-                    </Table>
-                </TableContainer>
-            )}
+            {({
+                rows,
+                headers,
+                getHeaderProps,
+                getTableProps,
+                getSelectionProps,
+                getRowProps,
+                getToolbarProps,
+                selectedRows,
+                getBatchActionProps
+            }) => {
+                const batchActionProps = getBatchActionProps();
+
+                return (
+                    <TableContainer
+                        title='Pending Group Application'
+                        description='This is a list of all the group application for this semester.'
+                    >
+                        <TableToolbar {...getToolbarProps()}>
+                            <TableBatchActions {...batchActionProps}>
+                                <TableBatchAction
+                                    kind='danger'
+                                    tabIndex={batchActionProps.shouldShowBatchActions ? 0 : -1}
+                                    renderIcon={Close}
+                                    onClick={batchActionClick(selectedRows)}
+                                >
+                                    Reject
+                                </TableBatchAction>
+                            </TableBatchActions>
+                        </TableToolbar>
+                        <Table {...getTableProps()}>
+                            <TableHead>
+                                <TableRow>
+                                    <TableSelectAll {...getSelectionProps()} />
+                                    {headers.map((header, i) => (
+                                        <TableHeader key={i} {...getHeaderProps({ header })}>
+                                            {header.header}
+                                        </TableHeader>
+                                    ))}
+                                </TableRow>
+                            </TableHead>
+                            <TableBody>
+                                {rows.map((row, i) => (
+                                    <TableRow key={i} {...getRowProps({ row })}>
+                                        <TableSelectRow {...getSelectionProps({ row })} />
+                                        {row.cells.map(cell => (
+                                            <TableCell key={cell.id}>{cell.value}</TableCell>
+                                        ))}
+                                    </TableRow>
```

Comment:
`SmsDatatable` should help simplify this with `SimpleRow`. A small caveat to using a custom component for the rows, it becomes nested so we need to provide the `SimpleRow` with a `key` prop. There should already be an example in the system doing this